satpy/readers/tropomi_l2.py

```diff
@@ -158,6 +165,26 @@ def _rename_dims(self, data_arr):
             dims_dict['scanline'] = 'y'
         return data_arr.rename(dims_dict)
 
+    def prepare_geo(self, bounds_data):
+        # lat/lon bounds are ordered in the following way:
+        # 3----2
+        # |    |
+        # 0----1
+        # Extend longitudes and latitudes with one element to support "pcolormesh"
+        # Create the bottom array
+        bottom = np.hstack([bounds_data[:, :, 0], bounds_data[:, -1:, 1]])
+        # Create the top array
+        top = np.hstack([bounds_data[-1, :, 3], bounds_data[-1, -1, 2]])
+        # Stack vertically
+        dest = np.vstack([top, bottom])
+        # Convert to DataArray
+        dest = xr.DataArray(dest,
```

Comment:
Is `dest` a dask array here? Sometimes stacking operations can produce non-dask results.

Reply:
You're right. I convert it to dask array now.